File: auturbo_2018_spring/auturbo_2018_spring_week3/calculator/src/number_accumulator.cpp
// Author: (AuTURBO) Leon Ryuwoon Jung 

#include "ros/ros.h" 
#include "std_msgs/Int32.h"

class NumberAccumulator
{
 public:
  NumberAccumulator()
  {
    sub_number = n.subscribe("/number", 1, &NumberAccumulator::cbGetNumber, this);
  }

  void cbGetNumber(const std_msgs::Int32 msg_num)
  {
    int number = msg_num.data;

    fnAccumulate(number);
  }

  void fnAccumulate(int number)
  {
    i = i + number;

    ROS_INFO("accumulated number : %d", i);
  }

 private:
  ros::NodeHandle n;

  ros::Subscriber sub_number;

  int i = 0;
};

int main(int argc, char **argv)
{
  ros::init(argc, argv, "number_accumulator");

  NumberAccumulator numberAccumulator;

  ros::spin();

  return 0;
}

File: auturbo_2018_spring/auturbo_2018_spring_week3/calculator/nodes/number_accumulator.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

# Author: (AuTURBO) Leon Ryuwoon Jung 

import rospy

from std_msgs.msg import Int32

class NumberAccumulator():
    def __init__(self):
        self.sub_number = rospy.Subscriber('/number', Int32, self.cbGetNumber, queue_size = 1)

        self.i = 0

    def cbGetNumber(self, msg_num):
        number = msg_num.data

        self.fnAccumulate(number)

    def fnAccumulate(self, number):
        self.i = self.i + number

        rospy.loginfo('accumulated number : %d', self.i)

    def main(self):
        rospy.spin()

if __name__ == '__main__':
    rospy.init_node('number_accumulator')
    node = NumberAccumulator()
    node.main()